src/main/java/seedu/planner/logic/commands/module/ModuleAddCommand.java

```diff
@@ -80,19 +88,27 @@ public CommandResult execute(Model model) throws CommandException {
         // TODO: have an option to check globally (across all timetables) to prevent duplicate enrollments
         // NOTE: Multiple enrollments of the same module code in different timetables is intended behaviour,
         //       since you can retake modules under some circumstances.
-        if (model.hasEnrollment(moduleCode)) {
-            throw new CommandException(generateDuplicateMessage(moduleCode));
+        List<String> messages = new ArrayList<>();
+        for (ModuleCode moduleCode : moduleCodes) {
+            if (model.hasEnrollment(moduleCode)) {
+                // throw new CommandException(generateDuplicateMessage(moduleCode));
+                messages.add(generateDuplicateMessage(moduleCode));
+                continue;
+            }
+
+            // Check if module exists in module database
+            Module module = ModuleUtil.getModuleWithCode(moduleCode);
+            if (module == null) {
+                // throw new CommandException(generateModuleDoesNotExists(moduleCode));
+                messages.add(generateModuleDoesNotExists(moduleCode));
+                continue;
+            }
+
+            Enrollment enrollment = new Enrollment(moduleCode, Optional.empty(), module.getModuleCredit());
+            model.addEnrollment(enrollment);
+            messages.add(generateSuccessMessage(moduleCode));
```

Comment:
I had previously tested my implementation of ModuleAddCommand.
Please do not alter my implementation of ModuleAddCommand or ModuleRemoveCommand (or any related classes, such as the corresponding parsers)

Reply:
As spoken, let's test it out and see if it works fine. No point being sceptical when we can improve user experience.